ogr/ogrsf_frmts/miramon/mm_wrlayr.c

```diff
@@ -0,0 +1,7038 @@
+    for (iElem = 0; iElem < hMiraMonLayer->nFinalElemCount; iElem++)
+    {
+        // Bounding box
+        FlushTMP.SizeOfBlockToBeSaved =
+            sizeof(pMMArcLayer->pArcHeader[iElem].dfBB.dfMinX);
+        FlushTMP.pBlockToBeSaved =
+            (void *)&pMMArcLayer->pArcHeader[iElem].dfBB.dfMinX;
+        hMiraMonLayer->OffsetCheck += FlushTMP.SizeOfBlockToBeSaved;
+        if (MMAppendBlockToBuffer(&FlushTMP))
+        {
+            if (pBuffer)
+                free_function(pBuffer);
+            return 1;
+        }
+        FlushTMP.pBlockToBeSaved =
+            (void *)&pMMArcLayer->pArcHeader[iElem].dfBB.dfMaxX;
+        hMiraMonLayer->OffsetCheck += FlushTMP.SizeOfBlockToBeSaved;
+        if (MMAppendBlockToBuffer(&FlushTMP))
+        {
+            if (pBuffer)
+                free_function(pBuffer);
+            return 1;
+        }
+        FlushTMP.pBlockToBeSaved =
+            (void *)&pMMArcLayer->pArcHeader[iElem].dfBB.dfMinY;
+        hMiraMonLayer->OffsetCheck += FlushTMP.SizeOfBlockToBeSaved;
+        if (MMAppendBlockToBuffer(&FlushTMP))
+        {
+            if (pBuffer)
+                free_function(pBuffer);
+            return 1;
+        }
+        FlushTMP.pBlockToBeSaved =
+            (void *)&pMMArcLayer->pArcHeader[iElem].dfBB.dfMaxY;
+        hMiraMonLayer->OffsetCheck += FlushTMP.SizeOfBlockToBeSaved;
+        if (MMAppendBlockToBuffer(&FlushTMP))
+        {
+            if (pBuffer)
+                free_function(pBuffer);
+            return 1;
+        }
+
+        // Element count: number of vertices of the arc
+        if (MMAppendIntegerDependingOnVersion(
+                hMiraMonLayer, &FlushTMP, &nUL32,
+                pMMArcLayer->pArcHeader[iElem].nElemCount))
+        {
+            if (pBuffer)
+                free_function(pBuffer);
+            return 1;
+        }
+
+        // Offset: offset of the first vertice of the arc
+        if (MMAppendIntegerDependingOnVersion(
+                hMiraMonLayer, &FlushTMP, &nUL32,
+                pMMArcLayer->pArcHeader[iElem].nOffset + nOffsetDiff))
+        {
+            if (pBuffer)
+                free_function(pBuffer);
+            return 1;
+        }
+        // First node: first node of the arc
+        if (MMAppendIntegerDependingOnVersion(
+                hMiraMonLayer, &FlushTMP, &nUL32,
+                pMMArcLayer->pArcHeader[iElem].nFirstIdNode))
+        {
+            if (pBuffer)
+                free_function(pBuffer);
+            return 1;
+        }
+        // Last node: first node of the arc
+        if (MMAppendIntegerDependingOnVersion(
+                hMiraMonLayer, &FlushTMP, &nUL32,
+                pMMArcLayer->pArcHeader[iElem].nLastIdNode))
+        {
+            if (pBuffer)
+                free_function(pBuffer);
+            return 1;
+        }
+        // Lenght of the arc
+        FlushTMP.SizeOfBlockToBeSaved =
+            sizeof(pMMArcLayer->pArcHeader[iElem].dfLenght);
```

Comment:
I'm just noticing that typo: dfLenght should be replaced globally to be dfLength  (t and h inverted)

Reply:
Oh! True! Done